crates/matrix-sdk/src/room/mod.rs

```diff
@@ -497,15 +521,15 @@ impl Room {
     /// This method will de-duplicate requests if it is called multiple times in
     /// quick succession, in that case the return value will be `None`. This
     /// method does nothing if the members are already synced.
-    pub async fn sync_members(&self) -> Result<Option<MembersResponse>> {
+    pub async fn sync_members(&self) -> Result<()> {
```

Comment:
Why the signature change for this and `request_members`? I couldn't find anything about it in the commit messages. Maybe you can add a changelog entry?

Reply:
Good question. On the one hand, no one was using the response here. On the other hand, if you were the second concurrent call for a given room, then the function wouldn't return anything; only the first caller would return something. Which makes for a very clunky API; the caller would have to synchronize the result, if they wanted to do anything with it.

Plus we have the `members` / `get_member` functions that internally call `sync_members` and return a more interesting thing then. Maybe we should make `sync_members` internal instead, like `request_members` already is?

Reply:
I think it makes sense that it continues to be public, but it would be nice to just call this out in the changelog.